src/eos-updater-poll-lan.c

```diff
@@ -0,0 +1,920 @@
+    }
+
+ out:
+  if (fail_key != NULL)
+    {
+      g_hash_table_steal_all (keys);
+      g_hash_table_insert (keys, (gchar *)fail_key, NULL);
+    }
+  return error;
+}
+
+static TxtRecordError
+get_unique_txt_records (gchar **txt_records,
+                        GHashTable **records,
+                        ...)
+{
+  gchar *key = NULL;
+  GHashTable *unique_keys = g_hash_table_new (g_str_hash, g_str_equal);
+  va_list args;
+
+  va_start (args, records);
+  for (;;)
+    {
+      key = va_arg (args, gchar*);
+
+      if (key == NULL)
+        break;
+      g_hash_table_insert (unique_keys, key, NULL);
+    }
+  va_end (args);
+
+  *records = unique_keys;
+  return fill_txt_records (txt_records, unique_keys);
+}
+
+static const gchar*
+get_unique_txt_record (gchar **txt_records,
+                       const gchar *key,
+                       TxtRecordError *txt_error)
+{
+  g_autoptr(GHashTable) records = NULL;
+  TxtRecordError error;
+
+  error = get_unique_txt_records (txt_records, &records, key, NULL);
+  if (txt_error != NULL)
+    *txt_error = error;
+  if (error != TXT_RECORD_OK)
+    return NULL;
+
+  /* the hash table does not manage the memory of its values, so we
+     don't have to steal them */
+  return g_hash_table_lookup (records, key);
+}
+
+#define EOS_TYPE_SERVICE_WITH_BRANCH_FILE eos_service_with_branch_file_get_type ()
+EOS_DECLARE_REFCOUNTED (EosServiceWithBranchFile,
+                        eos_service_with_branch_file,
+                        EOS,
+                        SERVICE_WITH_BRANCH_FILE)
+
+struct _EosServiceWithBranchFile
+{
+  GObject parent_instance;
+
+  EosAvahiService *service;
+  EosBranchFile *branch_file;
+
+  GDateTime *declared_download_time;
+  gchar *declared_sha512sum;
+};
+
+static void
+eos_service_with_branch_file_dispose_impl (EosServiceWithBranchFile *swbf)
+{
+  g_clear_object (&swbf->service);
+  g_clear_object (&swbf->branch_file);
+  g_clear_pointer (&swbf->declared_download_time, g_date_time_unref);
+}
+
+static void
+eos_service_with_branch_file_finalize_impl (EosServiceWithBranchFile *swbf)
+{
+  g_free (swbf->declared_sha512sum);
+}
+
+EOS_DEFINE_REFCOUNTED (EOS_SERVICE_WITH_BRANCH_FILE,
+                       EosServiceWithBranchFile,
+                       eos_service_with_branch_file,
+                       eos_service_with_branch_file_dispose_impl,
+                       eos_service_with_branch_file_finalize_impl)
+
+static EosServiceWithBranchFile *
+eos_service_with_branch_file_new (EosAvahiService *service)
+{
+  EosServiceWithBranchFile *swbf = g_object_new (EOS_TYPE_SERVICE_WITH_BRANCH_FILE, NULL);
+
+  swbf->service = g_object_ref (service);
+  swbf->branch_file = eos_branch_file_new_empty ();
+
+  return swbf;
+}
+
+static void
+eos_service_with_branch_file_get_from_lan_data (EosServiceWithBranchFile *swbf,
+                                                LanData *lan_data)
+{
+  g_set_object (&swbf->branch_file, lan_data->fetch_data->data->branch_file);
```

Comment:
Renamed it to even longer `eos_service_with_branch_file_set_branch_file_from_lan_data`. ;)